init an array of struct pointers from literal value

The following program `sample.go` triggers an unexpected error:

```go
package main

type T struct {
    name string
}

var tab = []*T{{
    name: "foo",
}, {
    name: "bar",
}}

func main() {
    println(len(tab))
    println(tab[0].name)
}
```

Expected result:
```console
$ go run ./sample.go
2
foo
```

Got:
```console
$ yaegi ./sample.go
./sample.go:8:2: undefined: name
```

--- a/_test/composite7.go
+++ b/_test/composite7.go
@@ -1,0 +1,20 @@
+package main
+
+type T struct {
+	name string
+}
+
+var tab = []*T{{
+	name: "foo",
+}, {
+	name: "bar",
+}}
+
+func main() {
+	println(len(tab))
+	println(tab[0].name)
+}
+
+// Output:
+// 2
+// foo

--- a/interp/type.go
+++ b/interp/type.go
@@ -1017,7 +1017,12 @@ func isInterface(t *itype) bool {
 func isStruct(t *itype) bool {
 	// Test first for a struct category, because a recursive interpreter struct may be
 	// represented by an interface{} at reflect level.
-	return t.cat == structT || t.TypeOf().Kind() == reflect.Struct
+	if t.cat == structT {
+		return true
+	}
+	rt := t.TypeOf()
+	k := rt.Kind()
+	return k == reflect.Struct || (k == reflect.Ptr && rt.Elem().Kind() == reflect.Struct)
 }
 
 func isBool(t *itype) bool { return t.TypeOf().Kind() == reflect.Bool }

